src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java

```diff
@@ -0,0 +1,584 @@
+                .withParams(ImmutableMap.of("stIdx", 0, "cnt", 50, "uri", "extInput:component")), JsonNode.class).at(RESULT_0_URI);
+        JsonNode compositeCount = this.doPost(AV_CONTENT_URI, getContentList
+                .withParams(ImmutableMap.of("stIdx", 0, "cnt", 50, "uri", "extInput:composite")), JsonNode.class).at(RESULT_0_URI);
+        JsonNode widiCount = this.doPost(AV_CONTENT_URI, getContentList
+                .withParams(ImmutableMap.of("stIdx", 0, "cnt", 50, "uri", "extInput:widi")), JsonNode.class).at(RESULT_0_URI);
+
+        populateContentInformation(hdmiCount, statistics, CONTENT_INFORMATION_TOTAL_HDMI_COUNT, CONTENT_INFORMATION_HDMI_PORT);
+        populateContentInformation(cecCount, statistics, CONTENT_INFORMATION_TOTAL_CEC_COUNT, CONTENT_INFORMATION_CEC_PORT);
+        populateContentInformation(componentCount, statistics, CONTENT_INFORMATION_TOTAL_COMPONENT_COUNT, CONTENT_INFORMATION_COMPONENT_PORT);
+        populateContentInformation(compositeCount, statistics, CONTENT_INFORMATION_TOTAL_COMPOSITE_COUNT, CONTENT_INFORMATION_COMPOSITE_PORT);
+        populateContentInformation(widiCount, statistics, CONTENT_INFORMATION_TOTAL_WIDI_COUNT, CONTENT_INFORMATION_WIDI_PORT);
+    }
+
+    /**
+     * Populate content information with "Unavailable" value for the contentCount if data is missing
+     * for the given source.
+     *
+     * @param json source
+     * @param statistics to add data to
+     * @param contentCountProperty to put content count information
+     * @param contentPortProperty to put port information to (port number: name)
+     */
+    private void populateContentInformation(JsonNode json, Map<String, String> statistics,
+                                            String contentCountProperty, String contentPortProperty){
+        if (!json.isMissingNode()) {
+            statistics.put(contentCountProperty, String.valueOf(json.size()));
+            json.forEach(jsonNode -> {
+                int portNumber = jsonNode.at(INDEX_URI).asInt() + 1;
+                statistics.put(contentPortProperty + portNumber, jsonNode.at(TITLE_URI).asText());
+            });
+        } else {
+            statistics.put(contentCountProperty, "Unavailable");
+        }
+    }
+
+    /**
+     * Retrieve data from {@link Constant#AUDIO_URI} endpoint with
+     * {@link LCDCommands#getSoundSettings} and {@link LCDCommands#getSpeakerSettings} payloads
+     *
+     * @param statistics to keep collected properties
+     * @param controls to keep controllable properties
+     * @throws Exception if any error occurs
+     * */
+    private void generateAudioStatistics(Map<String, String> statistics, List<AdvancedControllableProperty> controls) throws Exception {
+        ArrayNode soundSettings = (ArrayNode) this.doPost(AUDIO_URI, getSoundSettings, JsonNode.class).at(RESULT_0_URI);
+        ArrayNode speakerSettings = (ArrayNode) this.doPost(AUDIO_URI, getSpeakerSettings, JsonNode.class).at(RESULT_0_URI);
```

Comment:
making sure no copy/paste issue here, and both should use `AUDIO_URI` (just pointing it out, and I'm sure it is correct)

Reply:
yes, it is the same endpoint, different payloads though